dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -111,11 +172,22 @@ window.Dashboard = {
 			break;
 
 			case "SERVICE_UNAVAILABLE":
-				if (window.location.protocol === "http:" && this.config.INSTALL_CERT) {
-					this.__handleConfigReady();
-				} else {
+//				if (window.location.protocol === "http:" && this.config.INSTALL_CERT) {
+//					this.__handleConfigReady();
+//				} else {
 					this.__handleServiceUnavailable(event.status);
-				}
+//				}
+
+			case "CONTROLLER_UNAVAILABLE_FROM_HTTPS":
+				console.log("CONTROLLER_UNAVAILABLE_FROM_HTTPS", window.location.protocol);
+				// Controller isn't accessible via https. Redirect to http and try again.
+				window.location.href = window.location.href.replace("https", "http");
```

Comment:
We should add some logic here using `window.localStorage` to ensure this doesn't become an infinite redirect loop (should treat it as a `SERVICE_UNAVAILABLE` event if the redirect has already occurred in the last 2 seconds or so (play around with the timing to see what works best)).